CMakeLists.txt

```diff
@@ -106,6 +111,27 @@ if(${CMAKE_SYSTEM_NAME} MATCHES "Darwin")
     )
 endif()
 
+find_package(Check)
+if (CHECK_FOUND)
+  message(STATUS "check found")
+  add_library(Check::check SHARED IMPORTED)
+  set_target_properties(Check::check PROPERTIES
+    IMPORTED_LOCATION ${CHECK_LIBRARY}
+    INTERFACE_INCLUDE_DIRECTORIES ${CHECK_INCLUDE_DIR})
+else ()
+  set(check FALSE)
+  message(STATUS "check needed. The test suite process will be skipped.")
```

Comment:
Move this to the next `if`

Reply:
Could you please clarify what you mean and where this should be moved to?

Reply:
You have the message in the top-level `CMakeLists.txt` file, so you don't need it here